syntax = "proto3";

import "google/protobuf/empty.proto";
option go_package = ".;proto";

service UserFav {
  rpc GetFavPageList (UserFavFilterPageRequest) returns (UserFavListResponse); // 过滤收藏信息
  rpc CreateUserFav (CreateUserFavRequest) returns (google.protobuf.Empty); // 添加收藏
  rpc DeleteUserFav (DeleteUserFavRequest) returns (google.protobuf.Empty); // 取消收藏
  rpc GetUserFavDetail (DetailUserFavRequest) returns (google.protobuf.Empty); // 获取用户是否收藏
}

message UserFavFilterPageRequest {
  uint64 userId = 1;
  uint64 goodsId = 2;
}

message CreateUserFavRequest {
  uint64 userId = 1;
  uint64 goodsId = 2;
}

message UserFavResponse {
  uint64 userId = 1;
  uint64 goodsId = 2;
}

message UserFavDetailResponse {
  uint64 userId = 1;
  uint64 goodsId = 2;
}

message UserFavListResponse {
  uint64 total = 1;
  repeated UserFavResponse data = 2;
}

message DeleteUserFavRequest{
  uint64 userId = 1;
  uint64 goodsId = 2;
}

message DetailUserFavRequest{
  uint64 userId = 1;
  uint64 goodsId = 2;
}

